client_cli/src/main.rs

```diff
@@ -741,12 +744,8 @@ mod peer {
                 key,
                 metadata: Metadata(metadata),
             } = self;
-            submit(
-                RegisterBox::new(Peer::new(PeerId::new(&address, &key))),
-                cfg,
-                metadata,
-            )
-            .wrap_err("Failed to register peer")
+            let register_peer = RegisterBox::new(Peer::new(PeerId::new(&address, &key))).into();
```

Comment:
Everywhere else you are explicitly specify `Instruction` type.

Reply:
Indeed inferring works fine - removed type annotations for all cases.